addons/stock/report/report_stock_reception.py

```diff
@@ -248,6 +248,8 @@ def action_assign(self, move_ids, qtys, in_ids):
                     out.move_line_ids.move_id = new_out
                     assigned_amount = 0
                     for move_line_id in new_out.move_line_ids:
+                        if move_line_id.location_id != outs.location_id:
+                            continue
```

Comment:
Probably better to sort the `move_line_ids` putting the best candidate first rather than ignoring the non-perfect matches. Also the comparison should probably be with respect to the `potential_in.location_dest_id` rather than the `location_id` of the outs particularly because if the outs do not share the exact same source your code will crash. What about something like this ?
```suggestion
                    expected_locations = potential_ins.mapped('location_dest_id')
                    for move_line_id in new_out.move_line_ids.sorted(lambda ml: (ml.location_id not in expected_locations, ml.id):
```
Note that the condition of my proposition is written that way because `False < True`.